cmd/dbg/main.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/pflag"
+	"io/ioutil"
+	"net/http"
+	"os"
+)
+
+func main() {
+
+	//There aren't any flags yet, just arguments
+	var flags = pflag.NewFlagSet("", pflag.ExitOnError)
+	flags.Parse(os.Args)
+	args := flags.Args()
+
+	if len(args) == 2 && args[1] == "backends" {
+		backends()
+	} else if len(args) == 3 && args[1] == "backends" && args[2] == "list" {
+		backendsList()
+	} else if len(args) == 4 && args[1] == "backends" && args[2] == "get" {
+		backendsGet(args[3])
+	} else if len(args) == 2 && args[1] == "general" {
+		general()
+	} else if len(args) == 2 && args[1] == "conf" {
+		readNginxConf()
+	} else if len(args) == 1 {
+		info()
+	} else {
+		fmt.Println("Unknown command.")
+	}
+}
+
+func info() {
+	fmt.Println(`dbg is a tool for quickly inspecting the state of the nginx instance.
+Subcommands:
+
+- dbg backends             Output the dynamic backend information as JSON.
+- dbg backends list        Just list the names of all the backends.
+- dbg backends get <NAME>  Output the backend information only for the backend that has this name.
+- dbg general              Output the other dynamic information as JSON.
+- dbg conf                 Dump the contents of /etc/nginx/nginx.conf`)
+}
+
+func backends() {
+	//Get the backend information from the nginx instance
+	body, requestErr := makeRequest("http://127.0.0.1:18080/configuration/backends")
```

Comment:
Instead of `http://127.0.0.1:18080/configuration/backends` being hardcoded here, it can be a global variable.

Also `18080` should be obtained be user configurable.

Reply:
This port was disabled in [3684](https://github.com/kubernetes/ingress-nginx/pull/3684)